library-best-practices/glossary.md

```diff
@@ -66,3 +72,61 @@ Java Dependency Glossary
 - **Linkage-matchable version** (said of a particular version of A in relation
   to all versions of B): There exists some version of B such that the version of
   A and the version of B are linkage-compatible.
+
+
+### Static Linkage Checking
+
+- **Static linkage**: a reference from a class to another class that is
+  statically found in the Java class file of the former class.
+
+- **Static linkage error**: a linkage error for a static linkage.
+
+- **Static linkage check**: a process to identify the existence of _static
+  linkage conflicts_ for a given classpath, by scanning Java classes and
+  verifying the availability in the classpath.
+
+- **Class usage graph**: a possibly cyclic directed graph of references between classes
+  in the classpath. The nodes (vertices) of the graph correspond to
+  Java classes and the directed edges are references between classes.
+  For example, when 'Class A' has reference to 'Class B' on calling a method X,
+  the class usage graph holds an edge between two nodes:
+
+  ```
+  [Class A] --(method X of class B)-> [Class B]
+  ```
+
+  In this case, 'Class A' is called the _source class_ of the reference and
+  'Class B' is called the _destination class_.
+
+  In general, there can be multiple (parallel) edges between two nodes when
+  a class references two or more methods and fields of another class.
+  Self-loops (references between the same class) are possible and
+  common. However, it is safe to omit such references during static linkage checks,
+  because the Java compiler ensures that the self-loop references are linkage compatible.
+
+- **A reference**: in the class usage graph, the relationship between two 
```

Comment:
```suggestion
- **Reference**: in the class usage graph, the relationship between two 
```

Reply:
This sentence doesn't actually define a reference, it only says that there are sub-types. Could we give it a self-contained, one sentence definition?

Reply:
Added. What do you think?